### ISSUE TITLE
flash-attention wasn't compiled?

### ISSUE THREAD

**Issue by Donzox** [dicksondickson/ComfyUI-Clean-Install#4]
`C:\ComfyUI\comfy\ldm\modules\attention.py:407: UserWarning: 1Torch was not compiled with flash attention. (Triggered internally at C:\actions-runner\_work\pytorch\pytorch\builder\windows\pytorch\aten\src\ATen\native\transformers\cuda\sdp_utils.cpp:555.)
  out = torch.nn.functional.scaled_dot_product_attention(q, k, v, attn_mask=mask, dropout_p=0.0, is_causal=False)`

I ran `flash-attn-clone-source-compile-stable-torch.bat` line by line and build twice.

CUDA 12.6, 2.4.1+cu124.
I followed the instructions and used the lasted version of CUDA.

**Comment by dicksondickson** [comment]
This seems to be a pytorch or flash attention problem. From what I gathered, it has to deal with how Pytorch calls flash attention. Importing flash attention via python on windows works. This is something the flash attention devs needs to fix and make it work in windows.

**Comment by Donzox** [comment]
> This seems to be a pytorch or flash attention problem. From what I gathered, it has to deal with how Pytorch calls flash attention. Importing flash attention via python on windows works. This is something the flash attention devs needs to fix and make it work in windows.

So if downgrade to CUDA 12.4 and rebuild flash attention it will still have a problem?

**Comment by dicksondickson** [comment]
Its not a cuda problem. The problem goes away if you downgrade pytorch to 2.2.

**Comment by Donzox** [comment]
`** Python version: 3.11.2 (tags/v3.11.2:878ead1, Feb  7 2023, 16:38:35) [MSC v.1934 64 bit (AMD64)]
** Python executable: C:\ComfyUI\venv\Scripts\python.exe
** ComfyUI Path: C:\ComfyUI
** Log path: C:\ComfyUI\comfyui.log

Prestartup times for custom nodes:
   0.0 seconds: C:\ComfyUI\custom_nodes\rgthree-comfy
   1.1 seconds: C:\ComfyUI\custom_nodes\ComfyUI-Manager

Total VRAM 16376 MB, total RAM 32672 MB
** pytorch version: 2.2.0+cu121**
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 4070 Ti SUPER : cudaMallocAsync
Using pytorch cross attention`

Downgrade to 2.20 and I still get this message 
`C:\ComfyUI\comfy\ldm\modules\attention.py:407: UserWarning: 1Torch was not compiled with flash attention. (Triggered internally at ..\aten\src\ATen\native\transformers\cuda\sdp_utils.cpp:263.)
  out = torch.nn.functional.scaled_dot_product_attention(q, k, v, attn_mask=mask, dropout_p=0.0, is_causal=False)`

**Comment by dicksondickson** [comment]
Not sure why you are still getting that error. If you really want to get flash-attention to work with comfy, use linux. I have a dedicated workstation that runs popos / ollama / comfyui / flash attention and no errors but the performance gains is negligible .

**Comment by Donzox** [comment]
I'm not sure also, I just uninstalled the flash attention build `pip uninstall flash-attn` and still got the error after the model loads. I'll do a fresh install on a different drive and see if it'll then.

**Comment by dicksondickson** [comment]
Please do report back your findings. The only I got it to work was to use PopOS as my OS.

**Comment by Donzox** [comment]
Unfortunately the error still occurs. I did a fresh venv install on my second drive then the portable build from comfyui github on my third drive. There a plenty of others that encountered this issue from what I've looked up. I might take your advice and use linux.
